### ISSUE TITLE
Ignoring patches from dependencies doesn't work

### ISSUE THREAD

**Issue by dmsmidt** [cweagans/composer-patches#112]
I'm trying to ignore a patch defined in the composer.json of the Drupal distribution 'Open Social'. 

The setup mentioned in the readme, doesn't seem to work.

This part of the code seems never to be used:
```
 if (isset($extra['patches'])) {
          if (isset($patches_ignore[$package->getName()])) {
            foreach ($patches_ignore[$package->getName()] as $package => $patches) {
              if (isset($extra['patches'][$package])) {
                $this->io->write('<info>ignore patches each:' . implode(',', $patches) . '</info>');
                $extra['patches'][$package] = array_diff($extra['patches'][$package], $patches);
              }
            }
          }
          $this->patches = array_merge_recursive($this->patches, $extra['patches']);
        }
```

And in the following part regardless of ignore definitions dependency patches are added.

```
    foreach ($this->installedPatches as $patches) {
```

**Comment by danepowell** [comment]
I've noticed the same thing... I'm not able to get the patches-ignore functionality to work with Lightning.

**Comment by danepowell** [comment]
A couple of notes from Sam Mortenson:

1) Ignored patches don't support inheritance--i.e. if Lightning were to ignore an upstream patch, your sub-profile based on Lightning would not automatically ignore that patch.
2) We've seen caching problems with patches-ignore and patches where composer update doesn't re-apply or re-ignore patches, so my team has to clear composer cache often before building.
3) If neither of the above help, try evaluating one of the referenced PRs

**Comment by dmsmidt** [comment]
@danepowell 
1. That is correct, only the root composer.json is used for ignoring patches.
2. I've had similar problems.
3. It would  be great if you could test #118. Note that there are still cases that don't get caught, even with #118. See that issue for details.

**Comment by seanhamlin** [comment]
Also came across this issue with composer-patches. I am trying to ignore a patch from acquia/lightning, and no matter what I try, I cannot get this to work.

**Comment by alexdesignworks** [comment]
@cweagans this is still an issue in `1.6.3`. This broken functionality does not allow to set `"composer-exit-on-patch-failure": true,` as CI builds will fail for any projects relying on Lightning, as there is a patch in Lightning that does not apply.

**Comment by cweagans** [comment]
@alexdesignworks I think this is the wrong issue for that. Maybe you're looking for https://github.com/cweagans/composer-patches/issues/170?

**Comment by mikeshiyan-lc** [comment]
Any progress here?

Maybe https://github.com/cweagans/composer-patches/pull/118 can be simplified for now to get accepted and committed. Simply removing intermediate package (lightning, for example) should work.

**Comment by brantwynn** [comment]
I am currently finding that this mostly works, however patches are not re-ignored when running `composer update`  -- the patches from nested projects do work on `composer install`

In my case, I have `acquia/lightning` very deeply nested inside a dependency to `acquia/df` and it works on the initial install but again, not for updates. 

When I try to run `composer update` I see an error that the patch I have listed in `patches-ignore`

I do not see this error on `composer install` because the patch is successfully ignored.

**Comment by frederickjh** [comment]
I commented on pull #118 that I tested this using the instructions in [first issue that reference in that pull request, in this comment.](https://github.com/acquia/blt/issues/1379#issuecomment-295868485) This was using the forked and patch repo.

This works but does not work when using a separate patches file. The `patches-ignore` section needs to be in the main composer.json in my test.

I also try going back to an unpatched  version using the syntax in the README with the `patches-ignore` section in the main composer.json but this does not work.
